client/components/remove-button/index.jsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import assign from 'lodash/object/assign';
+import classNames from 'classnames';
+import noop from 'lodash/utility/noop';
+import Gridicon from 'components/gridicon';
+
+export default React.createClass( {
+
+	displayName: 'RemoveButton',
+
+	propTypes: {
+		icon: React.PropTypes.string,
+		disabled: React.PropTypes.bool,
+		onClick: React.PropTypes.func
+	},
+
+	getDefaultProps() {
+		return {
+			icon: 'remove',
+			disabled: false,
+			compact: false,
+			scary: false,
+			onClick: noop
+		};
+	},
+
+	getIconAssignments(){
+
+	},
+
+	render() {
+		const element = 'button';
+		const buttonIcons = {
+			'remove': 		'cross',
+			'trash': 			'trash',
+			'disconnect':	'link-break',
+			'deactivate':	'cross',
+			'delete':			'trash',
+		};
+		const buttonClasses = classNames( this.props.className, {
+			'button-remove': true,
+			'is-compact': this.props.compact,
+			'is-scary': this.props.scary
+		} );
+
+		if( this.props.compact ){
+			var iconSize = 12;
+		} else {
+			var iconSize = 24;
+		}
```

Comment:
All `var` declarations should be at the top of the function scope.

We could rewrite this as `let iconSize = this.props.compact ? 12 : 24;` perhaps.